### PR TITLE
[18.05] Respect workflow_building_modes.ENABLED in workflow manager's to_dict.

### LINKED ISSUE #5955: Can't edit component attributes when using use_remote_user
I've discovered a small issue with editing workflows when Galaxy is configured with `use_remote_user = True`. The problem is that if a user is directed straight to the workflow editor without having previously visited the Galaxy home page, then while they can add and remove components, they can't edit the attributes of components already in the workflow as clicking on a component results in an empty right hand side panel.

image

As you can see we are using a slightly edited version of Galaxy, and this is based on 17.09, but I don't believe any of the changes we have made are responsible (I've actually backed out most of the changes during testing and the problem remains).

Clearly the proxy in front of this instance is working as the user is logged in correctly, but as I say unless they have visited the home page of the Galaxy instance this problem occurs. It is easy to replicate this using a private browsing session to first visit the workflow editor for an existing workflow and see that the attributes don't appear, then in another tab visit the home page, and then finally refresh the tab showing the editor, at which point everything works as intended.

I've tried to figure out what's going on but so far have had little success. There are no obvious errors in the console, or any network errors (i.e. no 404 or 403 responses). My guess is that something in the session isn't being set correctly if you don't first visit the home page. Unfortunately in our use case we don't want users going to any page but the workflow editor (hence the UI changes to remove access to the rest of Galaxy). I know we are using an older version of Galaxy but does anyone have any thoughts/suggestions as to what might be going wrong. I'm happy to poke around and try and provide a fix if someone can suggest where I should look.

### PR DESCRIPTION
**Pull request by dannon**
This fixes #5955.  Opening a second PR to dev with a bunch of refactoring I did while tracking this down.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
Thanks @dannon, that looks good to me!

**Comment by mvdbeek** [comment]
@dannon this is targeting dev, I guess we'd want 18.05 ? (Or maybe 18.01/17.09 ?)

**Comment by dannon** [comment]
@mvdbeek Yeah, my intent was 18.05 per the title and the commentary, heh.  Good catch, thanks!

File: lib/galaxy/managers/workflows.py [hunk 1]
```diff
@@ -475,7 +475,7 @@ def _workflow_to_dict_editor(self, trans, stored):
             # Get user annotation.
             annotation_str = self.get_item_annotation_str(trans.sa_session, trans.user, step) or ''
             config_form = None
-            if trans.history:
+            if trans.history or trans.workflow_building_mode == workflow_building_modes.ENABLED:
```
Comment (guerler) [inline lib/galaxy/managers/workflows.py]: This if-statement should probably be just removed, then we always provide the config form when requesting a dictionary for the workflow editor. Thanks for fixing this @dannon.

Comment (dannon) [reply lib/galaxy/managers/workflows.py]: Per the comment in the statement, we don't want the config form if it isn't a web request.  I assume this was at some point an intentional API optimization -- @jmchilton?

Comment (jmchilton) [reply lib/galaxy/managers/workflows.py]: I'm plus 1 on dropping the if, I doubt it was an optimization - I just don't think we should default to serving up config forms via the API by default (e.g. to bioblend). They are unstable and they shouldn't be consumed external to the UI client. I will say the correct way to handle that is probably not with ifs like this though - we need separate APIs for things only interesting to the client and not general purpose API consumption and short of that some sort of flag indicating the requestor - that is why I'm totally fine with dropping the if for this case.

Comment (dannon) [reply lib/galaxy/managers/workflows.py]: Alright, I'll go ahead and simplify that.  That's actually what I did at first, then figured it was a historical optimization.

Anyway, we should continue refactoring of this outside the 18.05 PR (whoever is interested -- not me).

**Comment by guerler** [comment]
Thanks for the quick fix @dannon.

**Pull request merged**